components/camera/camera.go

```diff
@@ -81,10 +82,20 @@ type Camera interface {
 	VideoSource
 }
 
+type (
+	// PacketCallback is the signature of the SubscribeRTP callback.
+	PacketCallback func(pkts []*rtp.Packet) error
+	// VideoCodecStreamSource is a source of video codec data.
+	VideoCodecStreamSource interface {
+		SubscribeRTP(r *StreamSubscription, packetsCB PacketCallback) error
+		Unsubscribe(r *StreamSubscription)
+	}
+)
+
 // A VideoSource represents anything that can capture frames.
 type VideoSource interface {
 	projectorProvider
-
+	VideoCodecStreamSource() (VideoCodecStreamSource, error)
```

Comment:
This method is needed to know if RTP passthrough is possible before committing to using the passthrough or GetImage approach.